### PR TITLE
New build matrices and configs

### PR DESCRIPTION
**Pull request by sgillies**
Python 2.7 and GDAL 1.11-2.2 are removed.

Pythons 3.7 and 3.8 and GDAL nightly are added.

### CODE REVIEW

**Comment by sgillies** [comment]
The python 3.6 + GDAL 3.0.2 build won't import https://ci.appveyor.com/project/sgillies/fiona/builds/28752008/job/fd5ns34lijaxmluf#L1329.

The python 3.7 and GDAL 3.0.2 build crashes early in the tests https://ci.appveyor.com/project/sgillies/fiona/builds/28752008/job/vkjdqkw77knt1304#L1342.

The python 3.8 build fails to install the wheel https://ci.appveyor.com/project/sgillies/fiona/builds/28752008/job/ackwh2xob74rqvt1#L1196.

**Comment by rbuffat** [comment]
> The python 3.8 build fails to install the wheel https://ci.appveyor.com/project/sgillies/fiona/builds/28752008/job/ackwh2xob74rqvt1#L1196.

https://wheel.readthedocs.io/en/stable/news.html
Maybe it helps upgrading to wheel==0.33.6

**Comment by sgillies** [comment]
Good news: on Travis we have Pythons 3.6-3.8 and GDAL 2.3-3.0 passing https://travis-ci.org/Toblerity/Fiona/builds/610129304.

Bad news: no builds using GDAL 3.0 or "3.1" (the GDAL master branch) pass on AppVeyor. I don't have enough Windows expertise to understand why. There may be hints in other projects.

File: appveyor.yml [hunk 1]
```diff
@@ -11,41 +11,71 @@ environment:
         # See: http://stackoverflow.com/a/13751649/163740
         CMD_IN_ENV: "cmd /E:ON /V:ON /C .\\appveyor\\run_with_env.cmd"
         GDAL_HOME: "C:\\gdal"
-        PYTHON_ARCH: "64"
+        PYTHONWARNINGS: "ignore:DEPRECATION::pip._internal.cli.base_command"
```
Comment (rbuffat) [inline appveyor.yml]: This was needed to suppress the pip Python 2.7 deprecation warnings. Currently, it is not needed anymore. I think it does no harm for now, but maybe it is better to remove it now before its purpose gets forgotten in a few years.

File: appveyor.yml [hunk 2]
```diff
@@ -11,41 +11,71 @@ environment:
...
 
-    matrix:
-        # The 4-digit number in the GISInternals archives is the MSVC version used to build
-        # the libraries. It does not need to match the version of MSVC used to build Python.
-        # https://en.wikipedia.org/wiki/Microsoft_Visual_C%2B%2B#Internal_version_numbering
```
Comment (rbuffat) [inline appveyor.yml]: I think this comment is useful. I think it was already missing in my PR, maybe I accidentally deleted it.

File: appveyor.yml [hunk 3]
```diff
@@ -113,7 +142,8 @@ build_script:
 
   # install the wheel
   - ps: python -m pip install --upgrade pip
-  - ps: python -m pip install (gci dist\*.whl | % { "$_" })
+  - ps: python -m pip install --no-deps --ignore-installed (gci dist\*.whl | % { "$_" })
```
Comment (rbuffat) [inline appveyor.yml]: Would it make sense to drop the --upgrade in .travis.yml and also use --no-deps --ignore-installed?

**Comment by rbuffat** [comment]
@sgillies  I think https://github.com/Toblerity/Fiona/pull/826 is missing

**Pull request merged**